contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -369,6 +376,164 @@ tsql_get_constraintdef(PG_FUNCTION_ARGS)
 
 PG_FUNCTION_INFO_V1(tsql_get_constraintdef);
 
+/*
+ * tsql_get_functiondef
+ *		Returns the complete "CREATE OR REPLACE FUNCTION ..." statement for
+ *		the specified function.
+ *
+ * Note: if you change the output format of this function, be careful not
+ * to break psql's rules (in \ef and \sf) for identifying the start of the
+ * function body.  To wit: the function body starts on a line that begins
+ * with "AS ", and no preceding line will look like that.
+ */
+Datum
+tsql_get_functiondef(PG_FUNCTION_ARGS)
+{
+	Oid			funcid = PG_GETARG_OID(0);
+	StringInfoData buf;
+	StringInfoData dq;
+	HeapTuple      proctup;
+	Form_pg_proc proc;
+	bool		isfunction;
+	Datum		tmp;
+	bool	   isnull;
+	const char *prosrc;
+	const char *name;
+	const char *nsp;
+	const char *nnsp;
+	int	oldlen;
+        int* typmod_arr = NULL;
+        int number_args;
+
+	initStringInfo(&buf);
+
+	/* Look up the function */
+	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(funcid));
+	if (!HeapTupleIsValid(proctup))
+		PG_RETURN_NULL();
+
+	proc = (Form_pg_proc) GETSTRUCT(proctup);
+	name = NameStr(proc->proname);
```

Comment:
this can't be done as we are not allocating extra space for it.  we’re just using proc-> proname. I have already added comment to address it in the new commit.